Wikipedia/Images/WMFImageController.swift

```diff
@@ -124,14 +123,54 @@ public class WMFImageController : NSObject {
         .then(mainImageBlock)
     }
 
-    /// MARK: - Simple Fetching
+    /**
+    Use the shared session to download an image directly to the receiver's disk cache.
+
+    - parameter url: The URL to download the image from.
+
+    - returns: A promise which is resolved after the data has been successfully written to disk.
+    */
+    public func downloadImageDataInBackground(url: NSURL) -> Promise<Void> {
+        if hasDataOnDiskForImageWithURL(url) {
+            // already done!
+            return Promise()
+        }
+
+        let (promise, fulfill, reject) = Promise<NSURL>.pendingPromise()
+
+        /*
+          FIXME: Using sharedSession to punt on refactoring this to work with background sessions (which require using
+          a delegate).
+        */
+        let task = NSURLSession.sharedSession()
```

Comment:
See my comment above, I don't think it would be trivial to customize SDWebImage to write directly to disk, but I'll to a quick peek and see what I can find.